### PR TITLE
added `apply_support` and `max_deflection` methods

### PR DESCRIPTION
**Pull request by jashan498**
#### Brief description of what is fixed or changed
`apply_support` applies Support at a particular position. Support can be `cantilever`(fixed support), `roller` or `hinge`. reactional load and moment exerted by the support are represented by `R_x` and `M_x` respectively where `x` is the position of applied support.

### CODE REVIEW

**Comment by jashan498** [comment]
ping @parsoyaarihant @moorepants

File: sympy/physics/continuum_mechanics/tests/test_beam.py [hunk 1]
```diff
@@ -219,3 +219,31 @@ def test_remove_load():
...
     assert b.load == 0
     assert b.applied_loads == []
+
+
+def test_apply_support():
+    E = Symbol('E')
+    I = Symbol('I')
+    b = Beam(4, E, I)
+    b.apply_support(0, "cantilever")
+    b.apply_load(20, 4, -1)
```
Comment (parsoyaarihant) [inline sympy/physics/continuum_mechanics/tests/test_beam.py]: Can you add tests for other types of supports also?

File: sympy/physics/continuum_mechanics/beam.py [hunk 2]
```diff
@@ -209,6 +210,68 @@ def bc_deflection(self):
     def bc_deflection(self, d_bcs):
         self._boundary_conditions['deflection'] = d_bcs
 
+    def apply_support(self, loc, type="cantilever"):
```
Comment (moorepants) [inline sympy/physics/continuum_mechanics/beam.py]: Should be `type='fixed'`.

File: sympy/physics/continuum_mechanics/beam.py [hunk 3]
```diff
@@ -209,6 +210,68 @@ def bc_deflection(self):
...
+
+        Parameters
+        ==========
+        loc : Sympifyable
+            Location of point at which support is applied.
+        type : String
+            Determines type of Beam support applied.
+            - For Simple pinned Support, type = "pin"
+            - For Cantilever Support, type = "fixed"
+            - For Rolling Support, type = "roller"
```
Comment (moorepants) [inline sympy/physics/continuum_mechanics/beam.py]: Cantilever is not a type of a support. I think this should be more specific about the degrees of freedom. Pin joint fixes 2 of 3 DoF, fixed 3 of 3 DoF, and roller, 1 of 3 DoF.

File: sympy/physics/continuum_mechanics/beam.py [hunk 4]
```diff
@@ -209,6 +210,68 @@ def bc_deflection(self):
...
+        Using the sign convention of upward forces and clockwise moment
+        being positive.
+
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import symbols
+        >>> E, I = symbols('E, I')
+        >>> b = Beam(30, E, I)
+        >>> b.apply_load(-8, 0, -1)
+        >>> b.apply_support(10, 'roller')
+        >>> b.apply_support(30, 'roller')
```
Comment (moorepants) [inline sympy/physics/continuum_mechanics/beam.py]: What is the difference in applying a supprt and a load? Didn't we add supports by applying unknown loads before? Is this method really necessary?

Comment (moorepants) [reply sympy/physics/continuum_mechanics/beam.py]: I guess it does let you track more carefully what the unknown loads are. So maybe worth it. Do you use that info internally?

File: sympy/physics/continuum_mechanics/beam.py [hunk 5]
```diff
@@ -209,6 +210,68 @@ def bc_deflection(self):
...
+            self.apply_load(reaction_load, loc, -1)
+            self.apply_load(reaction_moment, loc, -2)
+            self.bc_deflection = [(loc, 0)]
+            self.bc_slope.append((loc, 0))
+            self._supports.append((loc, type))
+
+    def support_locations(self):
+        """ Returns a list of tuples containing location of Support
+        and its type respectively."""
+        return self._supports
```
Comment (moorepants) [inline sympy/physics/continuum_mechanics/beam.py]: Should this be a property?

File: sympy/physics/continuum_mechanics/beam.py (hunk 4 above)
Comment (jashan498) [reply sympy/physics/continuum_mechanics/beam.py]: Yeah, we can add support before applying load too. But I didn't get you by which info is to be used internally?

File: sympy/physics/continuum_mechanics/beam.py (hunk 2 above)
Comment (jashan498) [reply sympy/physics/continuum_mechanics/beam.py]: done

**Pull request merged**